### PR TITLE
test: clean up internal testing utilities

### PR DESCRIPTION
**Pull request by crisbeto**
We have some internal proxies for all of the Jasmine functions, as well as some other helpers. This code hasn't been touched in more than 5 years, it can lead to confusion and it isn't really necessary since the same can be achieved using Jasmine.

These changes remove most of the code and clean up our existing unit tests.

### CODE REVIEW

File: test-main.js [hunk 1]
```diff
@@ -9,7 +9,7 @@
 // Tun on full stack traces in errors to help debugging
 Error.stackTraceLimit = Infinity;
 
-jasmine.DEFAULT_TIMEOUT_INTERVAL = 100;
```
Comment (crisbeto) [inline test-main.js:12]: Fun fact: this one wasn't actually doing anything. It was being overwritten to 3000 inside `testing_internal.ts`.

**Review by AndrewKushnir — APPROVED** [review]
Awesome cleanup, thanks a lot @crisbeto 👍

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:375533628:BASE:375533653:1621885053746:20a44500).

**Comment by AndrewKushnir** [comment]
FYI, presubmits (including a global one) went well, so we'd need to get a few more approvals and the PR should be ready to go!

File: packages/core/testing/src/async_test_completer.ts [hunk 2]
```diff
@@ -1,32 +0,0 @@
...
- * Copyright Google LLC All Rights Reserved.
- *
- * Use of this source code is governed by an MIT-style license that can be
- * found in the LICENSE file at https://angular.io/license
- */
-
-/**
- * Injectable completer that allows signaling completion of an asynchronous test. Used internally.
- */
-export class AsyncTestCompleter {
```
Comment (IgorMinar) [inline packages/core/testing/src/async_test_completer.ts:12]: omg, I'm so glad that this class is gone. Thank you so much @crisbeto !

**Review by IgorMinar — APPROVED** [review]
AWESOME work @crisbeto!

I found just one place where you widened the scope in which we type a variable to `any`. I don't want to nitpick this PR since removing all `any`s from the existing code is outside of the scope of this change, but we should be careful when doing these refactorings not to introduce new `any`s or widen the scope of existing ones.

Otherwise looks great! Thank you

Reviewed-for: global-approvers

**Comment by IgorMinar** [comment]
I've removed the last remaining reviewers because the changes are to a large degree mechanical and non-controversial, also they affect only tests. @crisbeto if you feel like you would like more reviewers feel free to add them.

**Comment by crisbeto** [comment]
The feedback has been addressed.

**Comment by crisbeto** [comment]
Merge assistance: we already ran a global presubmit which passed, but the CI status was reset after I rebased.

**Pull request closed without merging**